resource_mock/__init__.py

```diff
@@ -0,0 +1,112 @@
+import csv
+from os import getcwd
+
+from openpyxl import load_workbook
+from datetime import datetime
+from typing import Dict, List, NoReturn
+
+class Resource:
+    def __init__(self,path):
+        self.path = getcwd()+path
+        self.FPU = {}
+        self.FPUr = {}
+        self.GRAT = {}
+        self.fpu_to_barcode = {}
+        self.instruments = {}
+        self.mode = {}
+        self.LGS = {}
+        self.IFU = {'gs': {}, 'gn': {}}
+
+    def _load_fpu(self, name: str, site: str) -> Dict[str,str]:
+        barcodes = {}
+        ifu = {}
+        with open(f'{self.path}/GMOS{site.upper()}_{name}201789.txt') as f:
+            reader =  csv.reader(f, delimiter=',') 
+            for row in reader:
+                barcodes[row[0].strip()] = [i.rstrip() for i in row[2:]]
+                ifu[row[0].strip()] = row[1].strip()
+        return ifu, barcodes
+                
+    def _load_grat(self, site: str) -> Dict[str,str]:
+        out_dict = {}
+        with open(f'{self.path}/GMOS{site.upper()}_GRAT201789.txt') as f:
+            reader =  csv.reader(f, delimiter=',') 
+            for row in reader:
+                out_dict[row[0].strip()] = [i.rstrip() for i in row[1:]]
+        return out_dict
+
+    def _load_f2b(self, site: str) -> Dict[str,str]:
+        out_dict = {}
+        with open(f'{self.path}/gmos{site}_fpu_barcode.txt') as f:
+            
+            for row in f:
+                fpu, barcode = row.split()
+                out_dict[fpu] = barcode
+        return out_dict
+    
+    def _to_bool(self, b: str) -> bool:
+        return True if b == 'Yes' else False
+
+    def _excel_reader(self, sites: List[str]) -> NoReturn:
+        workbook = load_workbook(filename=f'{self.path}/2018B-2019A Telescope Schedules.xlsx')
+        for site in sites:
+            sheet = workbook[site]
+            self.instruments[site.lower()] = {row[0].value.strftime("%Y-%m-%d"): 
+                                              [c.value for c in row[3:] ] for row in sheet.iter_rows(min_row=2)}
+
+            self.mode[site.lower()] = {row[0].value.strftime("%Y-%m-%d"): row[1].value for row in sheet.iter_rows(min_row=2)}
+            self.LGS[site.lower()] = {row[0].value.strftime("%Y-%m-%d"): 
+                                      self._to_bool(row[2].value) for row in sheet.iter_rows(min_row=2)}
```

Comment:
You use `site.lower()` here multiple times, requiring three calls to calculate the same thing.
Change this to:
```
site_lower = site.lower()
self.mode[site_lower] = ...
...
```
so we only calculate it once and get rid of the repetition.

Same with `row[0].value.strftime("%Y-%m-%d")`:
```
date = row[0].value.strftime('%Y-%m-%d')
```
and then use `date` for the `dict` keys.

Reply:
I removed the comprehension for the dictionaries to apply this one